### ISSUE TITLE
Layouts with views using 'string pass-through'

### ISSUE THREAD

**Issue by laryn** [backdrop/backdrop-issues#2412]
I was trying to set up a layout that would apply to multiple views page displays that share the first part of the URL (e.g. 'path/one' 'path/two', etc.) using the Layout Path of `path/%` and designating the context variable as a `string pass-through`. This doesn't work -- should it?

Otherwise for this use, I'd have to create/duplicate a bunch of layouts with the full Views Page URL on each one, which is very redundant and wastes a lot of time and energy.

**Comment by docwilmot** [comment]
That wont work, unfortunately, because there wouldn't be any menu item called 'path/%'. 

For example, we know Node module creates a menu item called 'node/%' in `node_menu()`. So if you have a Layout with path set as 'node/%', when you view a node, Layout can resolve the node's path (`node/65` for example) and see that that node is actually "belongs" to the menu item `node/%`. It checks: "Is there a Layout for 'node/%'? Yes? Use that.

In your case you have several 'path/xxx' items but no 'path/%' has been defined either in `hook_menu()` or in Views. So Layout will check: this path is 'path/one'; does it belong to a menu item like 'path/%' No? Use default.

What we'd need Layout to do is a string check if a path doesnt resolve to a menu item wildcard. Like: 
- this path is 'path/one'; does it belong to a menu item like 'path/%' No?
- ok, remove the last item after the "/" and replace it with a "%" and check again.

This will obviously get complicated with more complex paths (eg match `path/one/test/two` to a layout set at `path/%/test/%`)
Just thinking out loud about the solution, not sure the consequences.

**Comment by Gormartsen** [comment]
I fixes this issue in contrib module https://github.com/backdrop-contrib/layout_wildcard with next code:
```
/**
 * Get a list of layouts that share a particular path.
 * @see layout_get_path_layout_names
 */
function layout_wildcard_get_path_layout_names($path) {
  $path_map = &backdrop_static(__FUNCTION__, array());

  $configs = layout_get_all_configs('layout');
  foreach ($configs as $layout_name => $config) {
    if (isset($config['path'])) {
      $path_map[$config['path']][] = $layout_name;
    }
    if (isset($config['paths']) && is_array($config['paths'])) {
      foreach($config['paths'] as $subpath) {
        $path_map[$subpath][] = $layout_name;  
      }
    }
  }
  # Fix issue #2412
  $ancestors = menu_get_ancestors(explode("/", $path));
  foreach($ancestors as $ancestor_path) {
    if(isset($path_map[$ancestor_path])) {
      return $path_map[$ancestor_path];
    }
  } 

  return isset($path_map[$path]) ? $path_map[$path] : array();
}
```

**Comment by laryn** [comment]
Closing this issue as there is now a method to do this in contrib, as @Gormartsen noted.
